android/quest/src/test/java/org/smartregister/fhircore/quest/ui/pin/PinViewModelTest.kt

```diff
@@ -58,12 +61,25 @@ class PinViewModelTest : RobolectricTest() {
   @Test
   fun testSetPinUiState() {
     val context = ApplicationProvider.getApplicationContext<Application>()
+    val applicationConfiguration = mockk<ApplicationConfiguration>()
+    val expectedMessage = "VHT will use this PIN to login"
+    val pinLoginState: MutableState<PinUiState> =
+      mutableStateOf(
+        PinUiState(
+          currentUserPin = "1245",
+          message = expectedMessage,
+          appName = "demo",
+          setupPin = true,
+          pinLength = 4, // set pinLength to a value greater than 0
+          showLogo = false
+        )
+      )
     every { secureSharedPreference.retrieveSessionPin() } returns "1245"
     every { secureSharedPreference.retrieveSessionUsername() } returns "demo"
-    pinViewModel.setPinUiState(true, context)
-    val pinUiState = pinViewModel.pinUiState.value
-    Assert.assertEquals("1245", pinUiState.currentUserPin)
-    Assert.assertTrue(pinUiState.setupPin)
+    pinViewModel.pinUiState.value = pinLoginState.value
+    Assert.assertEquals(expectedMessage, pinLoginState.value.message)
+    Assert.assertEquals("1245", pinLoginState.value.currentUserPin)
+    Assert.assertTrue(pinLoginState.value.setupPin)
```

Comment:
on closer inspection, this is not testing what we want it to. This is testing the value of pinLoginState, but we just set that above on line 66, so we know what the values will be. What we would like to test here are the values of the pinViewModel.

Relatedly, we need to make these tests cover the newly uncovered line. But, because PinUIState can not have a nullable message, we would have to set the pinLoginMessage in the app config to null and pass that through for processing, to verify that when it is null, we get the default value back.

Hope this helps, please let us know if you need more guidance on rewriting these tests.

Reply:
@pld I would require more guidance on rewriting this since its essential for the eCHIS project

Reply:
Schedule a meeting with me Sharon on Tuesday. I will assist you.

Reply:
@sharon2719 also wherever can we need to follow the convention below for test case naming as documented in - https://fhircore.smartregister.org/developer-setup/code-testing

`<methodName><ConditionUnderTest><ExpectedBehavior>`